### ISSUE TITLE
[DuckDNS] Using a fourth-level domain fails dns-01 challenge during txt record creation

### ISSUE THREAD

**Issue by KF5JWC** [home-assistant/addons#1791]
## The problem

When using a fourth-level domain name, addon attempts to update TXT record on the fourth-level domain, which fails.

DuckDNS only allows configuration of third-level domains, and that configuration is repeated at every further level \[[spec](https://www.duckdns.org/spec.jsp) § TXT Record API\].

eg. `third.duckdns.org`
```bash
$ curl 'https://www.duckdns.org/update?token=abc123&verbose=true&domains=third.duckdns.org&txt=hello'
OK
hello
UPDATED
```

vs. `fourth.third.duckdns.org`
```bash
$ curl 'https://www.duckdns.org/update?token=abc123&verbose=true&domains=fourth.third.duckdns.org&txt=hello'
KO
```

## Environment

- Add-on with the issue: DuckDNS
- Add-on release with the issue: 1.12.4
- Last working add-on release (if known): UNKN
- Operating environment (OS/Supervised): Homeassistant, the OS, whatever that's called.

## Problem-relevant configuration

```yaml
domains:
  - fourth.third.duckdns.com
```

## Traceback/Error logs

```txt
Processing fourth.third.duckdns.org
 + Signing domains...
 + Generating private key...
 + Generating signing request...
 + Requesting new certificate order from CA...
 + Received 1 authorizations URLs from the CA
 + Handling authorization for fourth.third.duckdns.org
 + 1 pending challenge(s)
 + Deploying challenge tokens...
KO + Responding to challenge for fourth.third.duckdns.org authorization...
 + Cleaning challenge tokens...
KO + Challenge validation has failed :(
ERROR: Challenge is invalid! (returned: invalid) (result: {
  "type": "dns-01",
  "status": "invalid",
  "error": {
    "type": "urn:ietf:params:acme:error:unauthorized",
    "detail": "Incorrect TXT record \"\" found at _acme-challenge.fourth.third.duckdns.org",
    "status": 403
  },
  "url": "https://acme-v02.api.letsencrypt.org/acme/chall-v3/1234/5678",  
  "token": "xxxxxxxxx"                                  
})
```

## Additional information

Suspect the resolution would be to continue passing the full domain to letsencrypt, but pass only third-level domain to DuckDNS.

eg. (awk)
```awk
BEGIN { FS="." }
/.duckdns.org$/ { print $(NF-2) ".duckdns.org" }
```

**Comment by KF5JWC** [comment]
This is not related to #1331

**Comment by KF5JWC** [comment]
Workaround: 

```yaml
domains:
  - third.duckdns.org
  - fourth.third.duckdns.org
aliases:
  - domain: fourth.third.duckdns.org
    alias: third.duckdns.org
```
 ```txt
 Processing third.duckdns.org with alternative names: fourth.third.duckdns.org
 + Checking domain name(s) of existing cert... changed!
 + Domain name(s) are not matching!
 + Names in old certificate: third.duckdns.org
 + Configured names: fourth.third.duckdns.org third.duckdns.org
 + Forcing renew.
 + Checking expire date of existing cert...
 + Valid till Apr 30 21:20:00 2021 GMT Certificate will not expire
(Longer than 30 days). Ignoring because renew was forced!
 + Signing domains...
 + Generating private key...
 + Generating signing request...
 + Requesting new certificate order from CA...
 + Received 2 authorizations URLs from the CA
 + Handling authorization for fourth.third.duckdns.org
 + Handling authorization for third.duckdns.org
 + Found valid authorization for third.duckdns.org
 + 1 pending challenge(s)
 + Deploying challenge tokens...
OK + Responding to challenge for fourth.third.duckdns.org authorization...
 + Challenge is valid!
 + Cleaning challenge tokens...
OK + Requesting certificate...
 + Checking certificate...
 + Done!
 + Creating fullchain.pem...
 + Done!
 ```

**Comment by KF5JWC** [comment]
Will test with #1785

**Comment by KF5JWC** [comment]
Resolved by #1785 

```
 + Configured names: fourth.third.duckdns.org
 + Forcing renew.
 + Checking expire date of existing cert...
 + Valid till May 10 12:14:42 2021 GMT Certificate will not expire
(Longer than 30 days). Ignoring because renew was forced!
 + Signing domains...
 + Generating private key...
 + Generating signing request...
 + Requesting new certificate order from CA...
 + Received 1 authorizations URLs from the CA
 + Handling authorization for fourth.third.duckdns.org
 + Found valid authorization for fourth.third.duckdns.org
 + 0 pending challenge(s)
 + Requesting certificate...
 + Checking certificate...
 + Done!
 ```
 + Creating fullchain.pem...
 + Done!
